packages/components/src/image_zoom/index.jsx

```diff
@@ -0,0 +1,152 @@
+import { useCallback, useMemo, useRef } from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { useEventListener } from 'nebenan-react-hocs/lib/use_event_listener';
+import Slider from '../slider';
+import styles from './index.module.scss';
+import {
+  useDoubleTapZoom,
+  useDrag,
+  useImage,
+  useImageView,
+  usePinchZoom,
+  usePreviewSize,
+  useStateControlledInput,
+} from './hooks';
+import { getOffsetFromMouse, getOffsetFromTouch, getScaledImageSize } from './utils';
+
+const ImageZoom = ({
+  src,
+  className: passedClassName,
+  children,
+  maxHeight,
```

Comment:
Why I had to add `maxHeight`: We previously talked about always wanting to show images in a 16:9 ratio but that's not really true. If you upload a picture, you want to show it in its original aspect ratio. Everything else wouldn't really make sense since it would give you quite a small 'view window' to see the image content. If we go with dynamic aspect ratio, we have to be able to limit the max height of the image - otherwise we prevent mobile users to properly use the gallery (they can't scroll down as easy as desktop users).

Open for suggestions for the implementation